### PR TITLE
Background reporter for non-sampled server stats

### PR DESCRIPTION
**Pull request by apg**
Commit messages provide a decent amount of detail. Would look to them.

### CODE REVIEW

File: lib/trashed/consts.rb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module Trashed
+  STATE, COUNTERS, GAUGES = 'trashed.state', 'trashed.counters', 'trashed.gauges'
```
Comment (schneems) [inline lib/trashed/consts.rb:2]: This assignment can cause issues. If you accidentally add a value without a constant then you'll silently assign one of the constants as an array.

Under the hood multiple assignment also allocates an array, if you're doing it a lot it's a performance concern, here it's not a perf hotspot, but wanted to mention.

File: lib/trashed/rack.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
 require 'trashed/resource_usage'
+require 'trashed/consts'
 
 module Trashed
   class Rack
-    STATE, TIMINGS, GAUGES = 'trashed.state', 'trashed.timings', 'trashed.gauges'
```
Comment (schneems) [inline lib/trashed/rack.rb:5]: Ahh, i see that was just moving a component.

File: lib/trashed/reporter.rb [hunk 3]
```diff
@@ -23,39 +23,68 @@ def report(env)
     end
 
     def report_logger(env)
-      timings = env[Trashed::Rack::TIMINGS]
+    end
+
+    def report_statsd(env)
```
Comment (schneems) [inline lib/trashed/reporter.rb]: Maybe `raise "must implement interface"` in empty methods

**Review by schneems — COMMENTED** [review]
Made a minor comment about raising in a super class for an unimplemented method.

**Comment by apg** [comment]
Closing this PR in hopes of getting it upstream.

**Pull request closed without merging**